src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -7450,6 +7450,16 @@ or SyntaxKind.PlusPlusToken
                 or SyntaxKind.MinusMinusToken
                 or SyntaxKind.MinusGreaterThanToken;
 
+            // Now look for another set of items that indicate that we're not an attribute, but instead are a collection
+            // expression misplaced in an invalid top level expression-statement. (like `[] + b`).  These are
+            // technically invalid. But checking for this allows us to parse effectively to then give a good semantic
+            // error later on. These cases came from: ParseExpressionContinued
```

Comment:
Should ParseExpressionContinued have a comment linking it to here? Same with ParsePostfixExpression above (more important to keep in sync as those are valid scenarios).

Reply:
sure.  i can add those reverse links.